tests/bsim/bluetooth/tester/compile.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Compile all the applications needed by the bsim tests in these subfolders
+
+#set -x #uncomment this line for debugging
+set -ue
+
+source ${ZEPHYR_BASE}/tests/bsim/compile.source
+
+: "${ZEPHYR_BASE:?ZEPHYR_BASE must be set to point to the zephyr root directory}"
+
+west twister -T ${ZEPHYR_BASE}/tests/bsim/bluetooth/tester/
+
+if [ "${BOARD_TS}" == "nrf5340bsim_nrf5340_cpuapp" ]; then
+app=tests/bluetooth/tester \
+    exe_name=bs_${BOARD_TS}_${app}_prj_conf \
+    conf_overlay=${ZEPHYR_BASE}/tests/bluetooth/tester/hci_ipc.conf \
+    sysbuild=1 compile
+else
+    app=tests/bluetooth/tester \
+        exe_name=bs_${BOARD_TS}_${app}_prj_conf sysbuild=1 compile
+fi
```

Comment:
It feels kind of messy to use both `twister` and the `compile` function, but I couldn't figure out any way to have twister compile an application in a different repository. If anyone knows, please let me know